### ISSUE TITLE
`disable_error_code`-`misc` would also mute `reveal_type`?

### ISSUE THREAD

**Issue by DanielYang59** [python/mypy#19840]
**Bug Report**

`disable_error_code`-`misc` would also mute `reveal_type`? 

**To Reproduce**

```python
from typing import reveal_type

a: int = 1
reveal_type(a)
```

A dummy `pyproject.toml`:

```toml
[project]
name = "test"
version = "0.0.0"
description = "hi"

[tool.mypy]
disable_error_code = ["misc"]

```

**Expected Behavior**

If I don't have `misc` disabled:
```console
➜  temp uv run mypy test_type.py
warning: No `requires-python` value found in the workspace. Defaulting to `>=3.13`.
test_type.py:4: note: Revealed type is "builtins.int"
Success: no issues found in 1 source file
```

**Actual Behavior**

```console
➜  temp uv run mypy test_type.py
warning: No `requires-python` value found in the workspace. Defaulting to `>=3.13`.
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 1.18.1
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used:  3.13.5

**Comment by wyattscarpenter** [comment]
Minimal repro:

Gist URL: https://gist.github.com/mypy-play/54777cc6d1644054a79e7f4efbefe57a
Playground URL: https://mypy-play.net/?mypy=master&python=3.12&flags=show-error-codes&gist=54777cc6d1644054a79e7f4efbefe57a

```py
from typing import reveal_type
reveal_type(1) #type: ignore[misc]
```

Relevant documentation: https://mypy.readthedocs.io/en/stable/error_code_list.html#miscellaneous-checks-misc

**Comment by JelleZijlstra** [comment]
I feel like this is working as intended. `reveal_type` uses the misc code, so if you disable misc it disables `reveal_type()`.

If you have some reason to disable the `misc` code, it might be better if instead we split out whatever error you got upset about so that you don't need to disable `misc`. `misc` includes a lot of things so it's not great to disable it.

**Comment by wyattscarpenter** [comment]
Maybe it would also make sense to give reveal_type an (invisible) error code of reveal-type. (And reveal_locals a code of reveal-locals.) Perhaps in addition to spinning off whatever @DanielYang59 wanted to silence with misc. (Is there any reason not to make every misc code its own code, besides not wanting to bother?)

**Comment by DanielYang59** [comment]
> I feel like this is working as intended. reveal_type uses the misc code, so if you disable misc it disables reveal_type().

From a design perspective I guess it makes sense, but as a user it feels a bit surprising — I would have expected `reveal_type` to be treated at a different level (e.g. “DEBUG”) rather than bundled with regular error checks (“ERROR”).

> If you have some reason to disable the misc code, it might be better if instead we split out whatever error you got upset about so that you don't need to disable misc. misc includes a lot of things so it's not great to disable it.

Yes I agree most rules in `misc` are useful and I have removed it from the exclusion list

**Comment by A5rocks** [comment]
> Maybe it would also make sense to give reveal_type an (invisible) error code of reveal-type

IMO it would make more sense to have a "note" error code, just as a compromise between maximal error codes (what you say) and implementation details.

I think what your PR proposes of using `None` is not great because it's possible (? is it?) that people want to suppress notes.

**Comment by wyattscarpenter** [comment]
@A5rocks I agree that my PR, making notes with no error code default to error code None, is probably the wrong tack. Initially I did this because I thought nobody would ever want to ignore a reveal_type (and also, we don't display error codes for notes anyway, which sort of suggests to the user that they don't have one...), but, you know, maybe they will. Might as well not stymie this hypothetical user by making blockers and notes the same code.

As a result of my investigations in that PR, I suspect that many (or perhaps just a small number of?) plugins to mypy generate errors of misc (probably by default) and also notes of misc (by default) for the same line, without parenting the notes to the error. It may be impossible given the current plugin API for them to parent notes to the relevant error (not sure about this). This means that changing the default code of uncoded notes (such as from misc to note or misc to None) would be a slightly annoying backwards-incompatible change; users would have to update their type-ignores for those lines. Mypy would be within its rights, given its guarantees, to make such a change, but it's a weak reason against making such a change. So, given the already weak reason to make such a change, I think perhaps we shouldn't change the default type of notes.
